update ci signal handbook to adress new report tool (#1795)
kubernetes/sig-release: release-team/role-handbooks/ci-signal/README.md

-   1/2 hour to 2 hours a day, every workday, checking failing tests and following up on failing test and flake issues.
-   Availability to attend the majority of Release Team (weekly) and Burndown meetings (daily during Code Freeze), subject to time zone appropriateness.
-   Ability to follow-up on test fixes during Code Freeze at arbitrary times to ensure rapid turnaround.
-   The time commitment becomes greater through the release cycle, peaking during Code Freeze. In the last week of Code Freeze, Shadows should expect to spend at least 10 hours and leads at least 20 hours.

### Additional Requirements for Shadows

The following requirements are all things that Leads should already have, and Shadows should acquire, with coaching, within the first weeks of the release cycle:

-   Have signed the contributor CLA for Kubernetes.
-   Become a Kubernetes org member. In many cases, this will be done with the sponsorship of the CI Signal Lead or Release Lead in the first week of the cycle.

>     -   The process to become one of these is in [our community membership ladder](https://github.com/kubernetes/community/blob/master/community-membership.md#requirements-for-outside-collaborators)

Reviewer: ```suggestion
    -   The process to become an org member is in [our community membership ladder](https://github.com/kubernetes/community/blob/master/community-membership.md#requirements-for-outside-collaborators)
```

-   General familiarity with our [test boards](https://testgrid.k8s.io/) and examining test results from automated tests.
-   Willingness and ability to follow-up with contributors about test failures, on Slack, email, Discuss, and SIG meetings, as appropriate.
-   Ability to file well-crafted Kubernetes issues, including labelling.
-   General knowledge of our SIG system and SIGs' areas of responsibility.

Additionally, the following qualifications make a candidate more suitable for the CI Signal team, but are not requirements:

-   Prior involvement with SIG Testing and the Test Infrastructure team.
-   Experience with automated testing, CI/CD, quality engineering, and/or quality assurance.

> -   Experience with Golang, as the CI Signal team reviews the [CI Signal reporting tool](#reporting-status) written in Golang.

Reviewer 2: Maybe more like
"A basic understanding of Golang to review and improve the CI Signal reporting tool"

### Additional Requirements for Leads

In addition to the above requirements for Shadows, most of which become prerequisites, CI Signal Leads must:

-   Have the ability to add a milestone to issues, so must be a member of the [milestone maintainers](https://github.com/orgs/kubernetes/teams/milestone-maintainers)
-   Have a working knowledge of our various test infrastructure tools, such as Testgrid, Triage, Spyglass, gubernator, Prow, and Tide.
-   Signal lead needs to understand what tests matter and generally how our testing infra is wired together.

>     -   He/she can ask previous CI Signal leads for advice
>     -   He/she can ask SIG-Testing for guidance

Reviewer: ```suggestion
    -   They can ask previous CI Signal leads for advice
    -   They can ask SIG-Testing for guidance
```

### During Code Freeze

-   Continue best practices from Burndown stage. Monitor master-blocking, master-informing, release-x.y-blocking, and release-x.y-informing dashboards on **daily basis**
-   Quickly escalate any new failures to release team and SIGs
-   Continuing to send [weekly CI signal report](#reporting-status)

### Exit Code Freeze

-   Once 1.x.0-rc.1 release branch is cut and master re-opens for next release PRs
    -   continue **release-x.y-blocking, and release-x.y-informing dashboards on daily basis**
    -   check the **scalability jobs on master-informing** as often as they run.
    -   you need not monitor master-blocking on a daily basis. It is, however, worth keeping an eye on master-blocking especially before risky cherry-picks make their way into the release branch
-   If tests have stabilized (they should have by now), you may stop sending the weekly CI report

### Release Cutting - Go or No-Go

Over the release cycle, the release engineering team will cut several releases (alpha, beta, rc), before it ends with a new major Kubernetes release. During the release cycle, tests fail and potentially prevent the cut of a new version.

> Therefore, the key objective is to create a picture if the release branch manager can starting the release process or if there is a major failure that needs to be solved first. To avoid a last-minute No-Go, the assigned CI Signal member aligns at least 2-3 days before the day of the release cut with the [corresponding release branch manager](<[https://github.com/kubernetes/sig-release/tree/master/releases](https://github.com/kubernetes/sig-release/tree/master/releases)>). Potentially blocking tests have to be followed up with high priority and being driven to be solved until release day. Having a short daily alignment telling the Go/No-Go indication should give an idea if this is going in the right direction. If the CI Signal member decides on a No-Go, the issues have to be addressed. A short heads up about what is blocking, what are the current actions to resolve this and, if possible, an estimation of how long it will take to be solved should be given to the release lead, branch manager and to the test corresponding SIG leads.

Reviewer: ```suggestion
Therefore, the key objective is to create a picture if the release branch manager can start the release process or if there is a major failure that needs to be solved first. To avoid a last-minute No-Go, the assigned CI Signal member aligns at least 2-3 days before the day of the release cut with the [corresponding release branch manager](<[https://github.com/kubernetes/sig-release/tree/master/releases](https://github.com/kubernetes/sig-release/tree/master/releases)>). Potentially blocking tests have to be followed up with high priority and being driven to be solved until release day. Having a short daily alignment telling the Go/No-Go indication should give an idea if this is going in the right direction. If the CI Signal member decides on a No-Go, the issues have to be addressed. A short heads up about what is blocking, what are the current actions to resolve this and, if possible, an estimation of how long it will take to be solved should be given to the release lead, branch manager and to the test corresponding SIG leads.
```

>>>>>>> a59b111 (update ci signal handbook to adress new report tool)
=======
Therefore, the key objective is to create a picture if the release branch manager can start the release process or if there is a major failure that needs to be solved first. To avoid a last-minute No-Go, the assigned CI Signal member aligns at least 2-3 days before the day of the release cut with the [corresponding release branch manager](<[https://github.com/kubernetes/sig-release/tree/master/releases](https://github.com/kubernetes/sig-release/tree/master/releases)>). Potentially blocking tests have to be followed up with high priority and being driven to be solved until release day. Having a short daily alignment telling the Go/No-Go indication should give an idea if this is going in the right direction. If the CI Signal member decides on a No-Go, the issues have to be addressed. A short heads up about what is blocking, what are the current actions to resolve this and, if possible, an estimation of how long it will take to be solved should be given to the release lead, branch manager and to the test corresponding SIG leads.

> >>>>>>> 4218e47 (Update release-team/role-handbooks/ci-signal/README.md)

Reviewer 3: Is this a committed merge conflict?

Author: Will fix it later, thanks

The appropriate response to a failing or flaking job on either a blocking or informing dashboard is to open an issue. The primary goal of opening an issue is to track the work that needs to be done to bring the job back to a healthy status while providing accurate signal. The format for a helpful and informative issue may vary depending on the type of job, the test(s) that are failing for the job, and the responsible parties for the job.

Before creating a new github issue, check if somebody else has already opened an issue. To filter already opened issues, [queries](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests#search-by-project-board) can be applied.

Example queries:

> ```bash

Reviewer 3: Is it bash for this kind of snippet?

Author: Not really but it highlights it and improves readability (IMHO)

-   [Link to the previous reports (old report tool in use)](https://docs.google.com/document/d/1Xesru5QkBxxl8WOirNcWI4ho3exqfQBGz6mYJeXOi3U/edit#heading=h.qs3772g2kexl)
-   CCing individual SIGs (kubernetes-sig-foo@googlegroups.com) that own failing tests help in getting their attention
-   [Sample Weekly CI Signal Report (old report tool in use)](https://groups.google.com/d/msg/kubernetes-dev/nO8aKz3tavE/-aBpStu4CQAJ)

Early and continuous reporting of test health proved greatly effective in rallying SIG attention to test failures and stabilizing the release much early in the release cycle.

### CI signal reporting tool

During the 1.23 release cycle, a reporting tool was [donated](https://github.com/kubernetes/release/pull/2309) and enhanced to improve the weekly ci signal report. Furthermore, a new dashboard was built.
Features are tracked on the [ci signal report 2.0 project board](https://github.com/orgs/kubernetes/projects/65). New feature requests can be added as notes to the "Under Discussion" column and discussed at a release engineering meeting.

> The tool is consists of two parts:

Reviewer 2: The tool consists of two parts:
